lib/dialects/mariadb/query.js

```diff
@@ -1,21 +1,21 @@
 'use strict';
 
-const Utils = require('../../utils');
-const logger = require('../../utils/logger');
-const debug = logger.getLogger().debugContext('sql:mariadb');
 const AbstractQuery = require('../abstract/query');
 const uuidv4 = require('uuid/v4');
 const sequelizeErrors = require('../../errors');
 const _ = require('lodash');
 const DataTypes = require('../../data-types');
+const Promise = require('../../promise');
 
 const ER_DUP_ENTRY = 1062;
 const ER_ROW_IS_REFERENCED = 1451;
 const ER_NO_REFERENCED_ROW = 1452;
 
+const debug = AbstractQuery.makeDebugContext('mariadb');
+
 class Query extends AbstractQuery {
   constructor(connection, sequelize, options) {
-    super(connection, sequelize, Object.assign({ showWarnings: false }, options));
+    super(connection, sequelize, debug, Object.assign({ showWarnings: false }, options));
```

Comment:
Why debug is passed to super.constructor, `AbstractQuery` is not using debug in constructor, third argument should be `options`.

Reply:
Uh oh.